### PR TITLE
Add FastAI example with `FastAIPruningCallback`

### PR DESCRIPTION
**Pull request by crcrpar**
__UPDATE__: According to [fastai's quickstart](https://docs.fast.ai/vision.data.html#Quickly-get-your-data-ready-for-training), we can easily change the combination of data augmentation algorithms like rotation and zooming in/out. How about handling data augmentation as a hyperparameter?

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
Maybe this PR is still WIP, but let me leave some comments.

1. Please exclude `fastai_simple.py` from the CI targets of `examples-python3.5` and `examples-python3.8`. `fastai` will not be installed for Python 3.5 and 3.8 (see [setup.py](https://github.com/optuna/optuna/blob/master/setup.py#L88)), and `fastai_simple.py` will cause errors during CI. Please edit [L305 for Python3.8](https://github.com/optuna/optuna/blob/master/.circleci/config.yml#L305) and [L331 for Python3.5](https://github.com/optuna/optuna/blob/master/.circleci/config.yml#L331).

2. [`pytorch_lightning_simple.py`](https://github.com/optuna/optuna/blob/master/examples/pytorch_lightning_simple.py#L11) uses the command-line option `--pruning` to turn on/off the pruning feature. I think it can also be applied to this example. What do you think of such an option?

**Comment by crcrpar** [comment]
> 1. Please exclude `fastai_simple.py` from the CI targets of `examples-python3.5` and `examples-python3.8`.

Thank you for pointing out.

> 2\. [`pytorch_lightning_simple.py`](https://github.com/optuna/optuna/blob/master/examples/pytorch_lightning_simple.py#L11) uses the command-line option `--pruning` to turn on/off the pruning feature. I think it can also be applied to this example. What do you think of such an option?

It sounds reasonable and I reflected it.

File: examples/fastai_simple.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+BATCHSIZE = 128
+EPOCHS = 10
+
+
+path = untar_data(URLs.MNIST_SAMPLE)
+data = ImageDataBunch.from_folder(path, bs=BATCHSIZE)
+
+
+def objective(trial: optuna.trial.Trial) -> float:
```
Comment (toshihikoyanase) [inline examples/fastai_simple.py]:
```suggestion
def objective(trial):
```
Please remove the type hint for simplicity (c.f.,  https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#dont-use-type-hint-in-examples).

File: examples/fastai_simple.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    $ python fastai_integration.py [--pruning]
+"""
+
+import argparse
+from functools import partial
+
+from fastai.vision import ImageDataBunch
+from fastai.vision import Learner
+from fastai.vision import URLs
+from fastai.vision import accuracy
```
Comment (toshihikoyanase) [inline examples/fastai_simple.py]: Please reorder the imports as suggested by flake8.

https://circleci.com/gh/optuna/optuna/25969
```
#!/bin/bash -eo pipefail
. venv/bin/activate
flake8 .
./examples/fastai_simple.py:22:1: H306: imports not in alphabetical order (fastai.vision.urls, fastai.vision.accuracy)
1     H306: imports not in alphabetical order (fastai.vision.urls, fastai.vision.accuracy)
```

File: examples/fastai_simple.py (hunk 1 above)
Comment (crcrpar) [reply examples/fastai_simple.py]: Ah, I forgot to remove this after the development. Thank you.

**Comment by hvy** [comment]
The changes basically LGTM so I could go ahead and merge this PR. Or do you want to add the data augmentation? The required changes seem rather minimal and maybe it'll make the example more interesting. What do you think @crcrpar, @toshihikoyanase ?

**Comment by hvy** [comment]
A small detail, maybe one which you can ignore, is that the top comment/description in the example makes it sound like the pruning feature is the main feature to demonstrate, but pruning is turned off through args by default. I'm wondering if we should rephrase the description a bit.

**Comment by toshihikoyanase** [comment]
@hvy I was not aware of the comment about data augmentation.

> The required changes seem rather minimal and maybe it'll make the example more interesting. 

I agree. I think we can work on it in this PR.

**Comment by hvy** [comment]
Great, @crcrpar could you possibly add the data augmentation as well?

**Comment by hvy** [comment]
Great, @crcrpar could you possibly add the data augmentation as well?

**Pull request merged**